### PR TITLE
Add fx4120 [v101] Sponsored tiles telemetry

### PR DESCRIPTION
**Pull request by lmarceau**
# Issue [fx4120](https://mozilla-hub.atlassian.net/browse/FXIOS-4120)
- Add specific sponsored tiles telemetry
- Remove other telemetry (that wasn't release yet) since didn't follow the requirement spec (i.e. sponsored_tile_impressions)
- Add new ping and tag yaml files into the project
- Add tests for new telemetry use cases

### CODE REVIEW

**Comment by lmarceau** [comment]
@dnarcese 
# Request for data collection review form

1. What questions will you answer with this data?
- How many users sees the sponsored tiles
- How many users clicks on a sponsored tile.
- On each impression and click, the tile id, position, the source, the advertiser and the url of the tile. This information is linked to a unique context id, without any client id.

2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements?
This will help us understand usage patterns around top sites tiles and the new sponsored tiles, as well as being able to provide information for partners.

3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
There are no other alternatives.

4. Can current instrumentation answer these questions?
No

5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.
Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.

Measurement Description | Data Collection Category | Tracking Bug #
-- | -- | --
The sponsored tile was shown | Category 2 - interaction data | https://mozilla-hub.atlassian.net/browse/FXIOS-4120
The sponsored tile was clicked | Category 2 - interaction data | https://mozilla-hub.atlassian.net/browse/FXIOS-4120
Reporting tile URL | Category 1 & 2 - technical & interaction data | https://mozilla-hub.atlassian.net/browse/FXIOS-4120
Advertiser name | Category 1 & 2 - technical & interaction data | https://mozilla-hub.atlassian.net/browse/FXIOS-4120
Tile identifier | Category 1 & 2 - technical & interaction data | https://mozilla-hub.atlassian.net/browse/FXIOS-4120
Context id | Category 1 & 2 - highly sensitive or clearly identifiable personal data | https://mozilla-hub.atlassian.net/browse/FXIOS-4120

6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
https://dictionary.telemetry.mozilla.org/apps/firefox_ios

7. How long will this data be collected? Choose one of the following:
This will expired in 6 months

8. What populations will you measure?
All channels, all locales, all countries

9. If this data collection is default on, what is the opt-out mechanism for users?
Default Glean SDK opt-out mechanism.

10. Please provide a general description of how you will analyze this data.
Glean

11. Where do you intend to share the results of your analysis?
Only on Glean, and with mobile teams.

12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
No third-party tools.

**Review by travis79 — APPROVED** [review]
This looks great from my perspective!

File: Client/pings.yaml
Comment (lmarceau) [reply Client/pings.yaml:18]: Ahh missed that one! Thanks!!

**Review by yoanarios — APPROVED** [review]
LGTM

**Comment by dnarcese** [comment]
@lmarceau why is `Context id` category 4?

**Comment by lmarceau** [comment]
@dnarcese I'm following the same categories as Android here: https://github.com/mozilla-mobile/fenix/pull/23945 and described in document https://docs.google.com/document/d/1aP8gbxnRp0iIqbsqIyuvTeNLZFz92OKJOiv9kQ0cKVc/edit#heading=h.sl9qon5uav6w

**Comment by dnarcese** [comment]
@travis79 can you explain more about what the context id is and why it is category 4?

**Comment by travis79** [comment]
> @travis79 can you explain more about what the context id is and why it is category 4?

Yes, I can try! 

Since this data cannot be linked to the normal Glean `client_id`, the `context_id` here serves the same purpose for this dataset, and acts as a unique identifier for an install. Just like the `client_id` is classified as category 4 data, so must the `context_id`. The `context_id` requires us as stewards to answer _yes_ to the review question regarding "Does this collection add any new [identifiers](https://github.com/mozilla/data-review/blob/main/appendix.md#identifiers)?", and would trigger an escalation for the data-review. 

Since this has already been escalated and approved once for Fenix, I would recommend speaking to the same representative from Trust & Legal who reviewed the `context_id` identifier on Fenix and I don't see any reason why an escalation for this would be delayed.

**Comment by dnarcese** [comment]
Al Lavassani has signed off on the data review by email 👍

**Comment by dnarcese** [comment]
r+
# Data Review Form

_1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way?_ 

Yes, telemetry is documented at https://dictionary.telemetry.mozilla.org/apps/firefox_ios

_2) Is there a control mechanism that allows the user to turn the data collection on and off?_ 

Yes, Firefox iOS provides data controls in the app settings

_3) If the request is for permanent data collection, is there someone who will monitor the data over time?_

This should be re-reviewed w/ Ashley Thomas every 6mo

_4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?_

Category 4

_5) Is the data collection request for default-on or default-off?_

Default on

_6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?_

Yes, and approved by trust & legal (Al Lavassani).

_7) Is the data collection covered by the existing Firefox privacy notice?_ 

Yes

_8) Does the data collection use a third-party collection tool?_ 

No

**Comment by lmarceau** [comment]
Failing test pass locally.

**Pull request merged**
